iam tutorials, fixed scrolly bugs, added rating buttons (#9617)
braze-inc/braze-docs: _docs/_developer_guide/in_app_messages/tutorials/conditionally_display.md

        let configuration = Braze.Configuration(apiKey: "YOUR_API_ENDPOINT", endpoint: "YOUR_API_KEY")
        configuration.logger.level = .debug

        // 2. Initialize Braze SDK instance
        let brazeInstance = Braze(configuration: configuration)
        AppDelegate.braze = brazeInstance

        // 3. Set up Braze In-App Message UI and delegate
        let inAppMessageUI = BrazeInAppMessageUI()
        inAppMessageUI.delegate = self
        brazeInstance.inAppMessagePresenter = inAppMessageUI

        return true
    }

    func inAppMessage(_ ui: BrazeInAppMessageUI,
                      displayChoiceForMessage message: Braze.InAppMessage) -> BrazeInAppMessageUI.DisplayChoice {
        if let showFlag = message.extras["custom_key"] as? String, showFlag == "true" {
            return .now
        } else {
            return .discard
        }
    }
}
```

```swift file=SampleApp.swift
import SwiftUI

@main
struct SampleApp: App {
  @UIApplicationDelegateAdaptor(AppDelegate.self) var appDelegate

  var body: some Scene {
    WindowGroup {
      YourView()
    }
  }
}
```

!!step
lines-AppDelegate.swift=5

> #### 1. Extend `BrazeInAppMessageUIDelegate`

Reviewer: ```suggestion
#### 1. Have your AppDelegate class conform to `BrazeInAppMessageUIDelegate`
```
"Extend" to me would suggest literally like telling someone to write
```
extension BrazeInAppMessageUIDelegate {
....
}
```

Author: Ah thanks, that's a semantic mistake on my part

> In your AppDelegate class, extend [`BrazeInAppMessageUIDelegate`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazeui/brazeinappmessageui/delegate). This will allow you to override it's `inAppMessage` method later on.

Reviewer: ```suggestion
In your AppDelegate class, extend [`BrazeInAppMessageUIDelegate`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazeui/brazeinappmessageui/delegate). This will allow you to override its `inAppMessage` method later on.
```

!!step
lines-AppDelegate.swift=12

#### 2. Turn on debugging (optional)

Turn on debugging while developing to make troubleshooting easier!

!!step
lines-AppDelegate.swift=19-21

#### 3. Set up Braze In-App Message UI and delegate

> `BrazeInAppMessageUI()` is the default UI renderer for IAMs. By setting it's delegate to `self`, it'll ensure it's called before any IAMs are invoked.

Reviewer: I also think "it's" is a little ambiguous here. I might go instead with something like
```suggestion
`BrazeInAppMessageUI()` is the default UI presenter for IAMs. Set your AppDelegate class instance as its delegate for handling IAM presentation events.
```
I'm sort of split on how much we should try and explain the delegation pattern here to devs who might be unfamiliar.

!!step
lines-AppDelegate.swift=26-33

#### 4. Override `DisplayChoice` with your logic

> Override [`BrazeInAppMessageUI.DisplayChoice](https://braze-inc.github.io/braze-swift-sdk/documentation/brazeui/brazeinappmessageui/displaychoice) with the checks you'd like to perform. There are 2 enumeration types we will use in this case:

Reviewer 2: ```suggestion
Override [`BrazeInAppMessageUIDelegate .inAppMessage(_:displayChoiceForMessage:)`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazeui/brazeinappmessageuidelegate/inappmessage(_:displaychoiceformessage:)-9w1nb) with the checks you'd like to perform. There are 2 enumeration types we will use in this case:
```

import com.braze.ui.inappmessage.InAppMessageOperation
import android.util.Log

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()

        // Enable verbose Braze SDK logs
        BrazeLogger.logLevel = Log.VERBOSE

        // Initialize Braze
        val brazeConfig = BrazeConfig.Builder()
            .setApiKey("YOUR-API-KEY")
            .setCustomEndpoint("YOUR-ENDPOINT")
            .build()
        Braze.configure(this, brazeConfig)

        registerActivityLifecycleCallbacks(
            BrazeActivityLifecycleCallbackListener()
        )

        // Set up in-app message listener
        BrazeInAppMessageManager.getInstance().setCustomInAppMessageManagerListener(object : IInAppMessageManagerListener {
            override fun beforeInAppMessageDisplayed(inAppMessage: IInAppMessage): InAppMessageOperation {
                // Check if we should show the message
                val shouldShow = inAppMessage.extras["should_display_message"] == "true"

                return if (shouldShow) {
                    // Show the message using Braze's UI
                    InAppMessageOperation.DISPLAY_NOW
                } else {

>                     // Discard the message

Reviewer 3: ```suggestion
                    // Discard the message (or we could also create our own UI using KVP values)
```

        brazeInstance.inAppMessagePresenter = inAppMessageUI

        return true
    }

    func inAppMessage(_ ui: BrazeInAppMessageUI,
                      displayChoiceForMessage message: Braze.InAppMessage) -> BrazeInAppMessageUI.DisplayChoice {
        if let showFlag = message.extras["should_display_message"] as? String, showFlag == "true" {
            return .now
        } else {
            return .discard
        }
    }
}
```

```swift file=SampleApp.swift
import SwiftUI

@main
struct SampleApp: App {
  @UIApplicationDelegateAdaptor(AppDelegate.self) var appDelegate

  var body: some Scene {
    WindowGroup {
      YourView()
    }
  }
}
```

!!step
lines-AppDelegate.swift=5

#### 1. Implement `BrazeInAppMessageUIDelegate`

In your AppDelegate class, implement the [`BrazeInAppMessageUIDelegate`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazeui/brazeinappmessageui/delegate). This will allow you to override its `inAppMessage` method later on.

!!step
lines-AppDelegate.swift=12

#### 2. Enable debugging (optional)

Enable debugging while developing to make troubleshooting easier!

!!step
lines-AppDelegate.swift=19-21

> #### 3. Set up Braze In-App Message UI and delegate

Reviewer 4: ```suggestion
#### 3. Set up Braze In-App Message UI and its delegate
```
